lib/python/picongpu/plugins/data/compare.py

```diff
@@ -0,0 +1,222 @@
+"""
+This file is part of the PIConGPU.
+
+Copyright 2017-2018 PIConGPU contributors
+Authors: Sophie Rudat
+License: GPLv3+
+"""
+from picongpu.plugins.emittance import EmittanceData
+from picongpu.plugins.energy_histogram import EnergyHistogramData
+
+import numpy as np
+import os
+import collections
+import json
+
+
+class CompareData(object):
+    """
+    Data Reader for the Energy Histogram Plugin.
+    """
+
+    def __init__(self, run_directory):
+        """
+        Parameters
+        ----------
+        simulation_directory : string
+            path to the run directory of PIConGPU
+            (the path before ``simOutput/``)
+        """
+        if run_directory is None:
+            raise ValueError('The run_directory parameter can not be None!')
+
+        self.run_directory = run_directory
+        self.json_suffix = "scan_ranges.json"
+
+    def get_data_path(self, species, variation, observable,
+                      species_filter="all"):
+        """
+        Return the path to the underlying data file.
+
+        Parameters
+        ----------
+        species : string
+            short name of the particle species, e.g. 'e' for electrons
+            (defined in ``speciesDefinition.param``)
+        species_filter: string
+            name of the particle species filter, default is 'all'
+            (defined in ``particleFilters.param``)
+
+        Returns
+        -------
+        A string with a path.
+        """
+        self.path_to_scan = os.path.split(os.path.split(
+                                          self.run_directory)[0])[0]
+        self.path_to_json = os.path.join(self.path_to_scan, self.json_suffix)
+        if not os.path.isfile(self.path_to_json):
+            raise IOError('The file {} does not exist.\n'
+                          'Did the simulation already run?'
+                          .format(self.path_to_json))
+        # take first value of other variable parameters
+        with open(self.path_to_json) as f:
+            range_dict = json.load(f)
+            filtered_range_dict = dict()
+            self.fix_param_dict = dict()
+            for name, attrs in range_dict.items():
+                filtered_range_dict[name] = attrs['values']
+                if ((name != variation) &
```

Comment:
the brackets are not needed (the outer *and* the inner ones can go).
the reason for that is [operator precedence](https://docs.python.org/3/reference/expressions.html#operator-precedence)

the `&` should be an `and` (otherwise it's a bitwise operation)